spec.html

```diff
@@ -37725,12 +37640,86 @@ <h1>
             1. Let _ySmaller_ be ! IsLessThan(_yString_, _xString_, *true*).
             1. If _ySmaller_ is *true*, return *1*<sub>𝔽</sub>.
             1. Return *+0*<sub>𝔽</sub>.
+          1. Return ? SortIndexedProperties(_obj_, _len_, _SortCompare_).
+        </emu-alg>
+        <emu-note>
+          <p>Because non-existent property values always compare greater than *undefined* property values, and *undefined* always compares greater than any other value, *undefined* property values always sort to the end of the result, followed by non-existent property values.</p>
+        </emu-note>
+        <emu-note>
+          <p>Method calls performed by the ToString abstract operations in steps <emu-xref href="#step-sortcompare-tostring-x"></emu-xref> and <emu-xref href="#step-sortcompare-tostring-y"></emu-xref> have the potential to cause _SortCompare_ to not behave as a consistent comparator.</p>
+        </emu-note>
+        <emu-note>
+          <p>The `sort` function is intentionally generic; it does not require that its *this* value be an Array. Therefore, it can be transferred to other kinds of objects for use as a method.</p>
+        </emu-note>
+
+        <emu-clause id="sec-sortindexedproperties" type="abstract operation">
+          <h1>
+            SortIndexedProperties (
+              _obj_: an Object,
+              _len_: a non-negative integer,
+              _SortCompare_: an Abstract Closure with two parameters,
+            ): either a normal completion containing an Object or an abrupt completion
+          </h1>
+          <dl class="header">
+          </dl>
+          <emu-alg>
+            1. Let _items_ be a new empty List.
+            1. Let _k_ be 0.
+            1. Repeat, while _k_ &lt; _len_,
+              1. Let _Pk_ be ! ToString(𝔽(_k_)).
+              1. Let _kPresent_ be ? HasProperty(_obj_, _Pk_).
+              1. If _kPresent_ is *true*, then
+                1. Let _kValue_ be ? Get(_obj_, _Pk_).
+                1. Append _kValue_ to _items_.
+              1. Set _k_ to _k_ + 1.
+            1. Let _itemCount_ be the number of elements in _items_.
+            1. [id="step-array-sort"] Sort _items_ using an implementation-defined sequence of calls to _SortCompare_. If any such call returns an abrupt completion, stop before performing any further calls to _SortCompare_ or steps in this algorithm and return that completion.
```

Comment:
We don't use the term "completion" on its own any more. We've removed its `<dfn>`.

```suggestion
            1. [id="step-array-sort"] Sort _items_ using an implementation-defined sequence of calls to _SortCompare_. If any such call returns an abrupt completion, stop before performing any further calls to _SortCompare_ or steps in this algorithm and return that Completion Record.
```

Reply:
Also, "or steps in this algorithm" is of dubious value. That's how returning works everywhere else in this document.

Reply:
> We don't use the term "completion" on its own any more.

Actually, the status quo has 2 or 3 such occurrences, including this one. Also, there's 7 or 8 occurrences of "completion value": what's the editorial stance on those?

Reply:
> Also, "or steps in this algorithm" is of dubious value. That's how returning works everywhere else in this document.

That phrase was added by PR #1585, apparently as part of a [suggestion](https://github.com/tc39/ecma262/pull/1585#discussion_r565492508) from @bakkot.

Reply:
We meant to eliminate "completion" or "completion value" as part of #2547. The editorial convention is to use "Completion Record" when talking about one generically or the terms "normal completion", "abrupt completion", etc when referring to specific kinds.

Reply:
Re "or steps in this algorithm", I don't think it needs to be (or should be) removed as part of this PR - I'd prefer to keep this a pure refactoring, and making other changes at the same time obscures those changes. @michaelficarra I'm fine with dropping the phrase in a separate PR if you feel inclined to make that change.

Reply:
+1 to separate PR.